atc/engine/step_metadata.go

```diff
@@ -35,5 +35,13 @@ func (metadata StepMetadata) Env() []string {
 		env = append(env, "BUILD_TEAM_NAME="+metadata.TeamName)
 	}
 
+	if metadata.ExternalURL != "" && metadata.TeamName != "" && metadata.PipelineName != "" && metadata.JobName != "" && metadata.BuildName != "" {
+		env = append(env, fmt.Sprintf("BUILD_URL=%s/teams/%s/pipelines/%s/jobs/%s/builds/%s", metadata.ExternalURL, metadata.TeamName, metadata.PipelineName, metadata.JobName, metadata.BuildName))
```

Comment:
Would it make sense to just have one env var, set to either the long URL if it's in a pipeline or the short URL if it's a one-off build? Having to use one or the other could make this difficult to integrate with.